plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -484,7 +484,13 @@ public ConnectorTableHandle getTableHandleForStatisticsCollection(ConnectorSessi
 
             handle = handle.withAnalyzePartitionValues(list);
             HivePartitionResult partitions = partitionManager.getPartitions(handle, list);
-            handle = partitionManager.applyPartitionResult(handle, partitions, Optional.empty());
+            handle = partitionManager.applyPartitionResult(
+                    handle,
+                    partitions,
+                    new Constraint(
+                            TupleDomain.all(),
+                            partitionValues -> true,
```

Comment:
use `Constraint.alwaysTrue`

if your intent is to pass a `Constraint#constraint` to trigger some underlying logic, this is quite a disguised way to do this.

Reply:
please switch to `Constraint.alwaysTrue()`

Reply:
But  for `Constraint.alwaysTrue` - can we pass a default predicate which evaluates to `true` ?